server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java

```diff
@@ -867,14 +896,18 @@ public boolean hasNormalizer() {
     private final IndexAnalyzers indexAnalyzers;
     private final int ignoreAboveDefault;
     private final int ignoreAbove;
+    private final BinaryFieldMapper offsetsFieldMapper;
```

Comment:
Do we actually need this? We only ever use `fullPath()` from it.

Reply:
Right, I think we can remove it. We can just keep track of offsetFieldName.

I think I wanted to share this field via `#iterator()`, so that this field is visible in other places like field caps, but I think this field should remain internal/hidden. Additionally `BinaryFieldMapper` uses a custom binary doc values field  that has no advantage for offsets.